.pipeline/scripts/release_notes_automation.py

```diff
@@ -0,0 +1,110 @@
+import argparse
+import os
+import re
+from datetime import datetime
+
+
+def read_file(file_name):
+    with open(file_name, 'r', encoding="utf8") as file:
+        return file.read()
+
+
+def write_file(file_name, data):
+    with open(file_name, 'w', encoding="utf8") as file:
+        file.write(data)
+
+unchanged_sections = [
+"""### 🚧 Known Issues
+
+-
+
+""",
+"""### 🔧 Compatibility Notes
+
+-
+
+""",
+"""### ✨ New Functionality
+
+-
+
+""",
+"""### 📈 Improvements
+
+-
+
+""",
+"""### 🐛 Fixed Issues
+
+- 
+"""]
+
+def remove_unchanged_sections(file, unchanged_sections):
+    for unchanged_section in unchanged_sections:
+        # if file contains unchanged_section, remove it
+        file = re.sub(unchanged_section, "", file)
+    return file
+
+def set_header(file, version):
+    date = datetime.today().strftime('%B %d, %Y')
+    # Replace the first line with: ## 5.2.0 - January 17, 2024
+    file = re.sub("^## .*", "## " + version + " - " + date, file)
+    return file
+
+def link_github_release(file, version):
+    old_github_release_link = "\[All Release Changes\]\(https://github.com/SAP/ai-sdk-java/releases/\)"
+    new_github_release_link = "[All Release Changes](https://github.com/SAP/cloud-sdk-java/releases/tag/rel%2F"+version+")"
+    file = re.sub(old_github_release_link, new_github_release_link, file)
+    return file
+
+
+
+releases_pattern = re.compile(r"^## ")
+def count_releases(filename):
+    count = 0
+    with open(filename, 'r', encoding="utf-8") as file:
+        for line in file:
+            if releases_pattern.match(line):
+                count += 1
+    return count
+
+def find_target_file(version):
+    # release-notes-X-to-Y.mdx with every 15 versions the index increases by 15 and stays the same for 15 versions
+    minor_version = int(version.split(".")[1])
+    index = minor_version // 15 * 15
+    return "release-notes-" + str(index) + "-to-" + str(index + 14) + ".mdx"
```

Comment:
(Question)

I don't understand this logic, isn't `index == minor_version` always?